### PR TITLE
Added MAUI tutorial/course from Microsoft

### PR DESCRIPTION
**Pull request by ikayz**
Added the course Build mobile and desktop apps with .NET MAUI from Microsoft with James Montemagno

## What does this PR do?
Add resource(s)

### Description
A free course from Microsoft on how to create Mobile and Desktop Apps using MAUI.

### Why is this valuable (or not)?
It's valuable for .Net developers who are looking into developing for Mobile and Desktop using a new modern C# framework

### How do we know it's really free?
It's listed as free on the Microsoft Learn website and can be accessed without registration or logging in.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Review by ikayz — COMMENTED** [review]
Changed the commit message to something more descriptive

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -339,6 +339,7 @@
 
 * [ASP.NET Core 2.2 & 3 REST API Tutorial](https://youtube.com/playlist?list=PLUOequmGnXxOgmSDWU7Tl6iQTsOtyjtwU&si=3XHU6BnqwxePAt54) - Nick Chapsas
 * [Building a microservice architecture with ASP.NET Core - Gill Cleeren - NDC London 2022](https://www.youtube.com/watch?v=SR53SKIUYPA) - Gill Cleeren
+* [Build mobile and desktop apps with .NET MAUI](https://learn.microsoft.com/en-us/training/paths/build-apps-with-dotnet-maui) - James Montemagno (Microsoft)
```
Comment (rng70) [inline courses/free-courses-en.md:342]: Swap line 342 and 341. These are sorted in alphabetical order. Swapping them will resolve the issue. For further information see [here](https://github.com/vhf/remark-lint-alphabetize-lists#remark-lint-alphabetize-lists-)

**Review by rng70 — CHANGES_REQUESTED** [review]

File: courses/free-courses-en.md (hunk 1 above)
Comment (ikayz) [reply courses/free-courses-en.md:342]: @rng70 thanks. Let me do that, I was wondering what the problem was and couldn't see it.

**Review by ikayz — COMMENTED** [review]
Swapped lines 341 and 342 to fix lint error

**Pull request merged**